Makefile

```diff
@@ -40,14 +43,19 @@ xtest:
 ta:
 	$(q)$(MAKE) -C ta CROSS_COMPILE="$(CROSS_COMPILE_TA)" \
 			  q=$(q) \
-			  O=$(out-dir)/ta \
+			  O=$(out-dir) \
 			  $@
 
 .PHONY: clean
 ifneq ($(wildcard $(TA_DEV_KIT_DIR)/host_include/conf.mk),)
 clean:
 	$(q)$(MAKE) -C host/xtest O=$(out-dir)/xtest q=$(q) $@
-	$(q)$(MAKE) -C ta O=$(out-dir)/ta q=$(q) $@
+ifneq ($O,)
+	$(q)bash -c 'cd host/xtest && [ ! -d $(O) ] || \
+			[ `stat --format=%i $(O)` == `stat --format=%i .` ] || \
+			rmdir --ignore-fail-on-non-empty  $(O)'
```

Comment:
Since we basically want to ignore any error, the whole `ifneq` block could be simplified to:
```
        rmdir $(O) 2>/dev/null; true
```